components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+		ctx = null;
+		if (log.isDebugEnabled()) {
+			log.debug("Tenant flow ended");
+		}
+	}
+
+	@POST
+	@Path("/inactive-policy")
+	@Produces("application/json")
+	public boolean addPolicy(Policy policy) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.addPolicy(policy);
+			response.setStatus(Response.Status.CREATED.getStatusCode());
+			if (log.isDebugEnabled()) {
+				log.debug("Policy has been added successfully.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@POST
+	@Path("/active-policy")
+	@Produces("application/json")
+	public boolean addActivePolicy(Policy policy) {
+
+		policy.setActive(true);
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint pap = policyManagerService.getPAP();
+			pap.addPolicy(policy);
+			response.setStatus(Response.Status.CREATED.getStatusCode());
+			if (log.isDebugEnabled()) {
+				log.debug("Policy has been added successfully.");
+			}
+			return true;
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Produces("application/json")
+	public Policy[] getAllPolicies() {
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			PolicyAdministratorPoint policyAdministratorPoint = policyManagerService.getPAP();
+			List<Policy> policies = policyAdministratorPoint.getPolicies();
+			return policyAdministratorPoint.getPolicies().toArray(new Policy[policies.size()]);
+		} catch (PolicyManagementException e) {
+			String error = "Policy Management related exception";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return null;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return null;
```

Comment:
As a best practice return an empty array as the default output of a method that is expected to return an array of elements.